packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -40,8 +1469,47 @@ String describeEnum(Object enumEntry) {
   return description.substring(indexOfDot + 1);
 }
 
-/// A mixin that helps dump string representations of trees.
-abstract class TreeDiagnosticsMixin {
+/// Returns a lower case hyphen separated version of a camel case name.
+///
+/// For example:
+///
+/// ```dart
+///
+/// main() {
+///   assert(toHyphenedName('deferToChild') == 'defer-to-child');
+///   assert(toHyphenedName('Monday') == 'monday');
+///   assert(toHyphenedName('monday') == 'monday');
+/// }
+/// ```
+String toHyphenedName(String word) {
```

Comment:
This method name should probably include the fact that it's going from camelCase, though. Otherwise people may think it's for general hyphenation.